tests/testthat/test-vroom_write.R

```diff
@@ -306,3 +306,23 @@ test_that("na argument modifies how missing values are written", {
   df <- data.frame(x = c(NA, "x"), y = c(1, 2))
   expect_equal(vroom_format(df, ",", na = "None"), "x,y\nNone,1\nx,2\n")
 })
+
+test_that("vroom_write() does not overwrite file when appending empty data frame", {
+  tf <- withr::local_tempfile()
+  data <- tibble::tibble(
+    a = "1",
+    b = "2",
+    c = "3"
+  )
+
+  vroom_write(data, file = tf, delim = ",")
+  vroom_write(data.frame(), file = tf, append = TRUE, delim = ",")
+
+  expect_equal(
+    vroom_lines(tf,
+      altrep = FALSE,
+      skip_empty_rows = FALSE
```

Comment:
I can't see why we need to specify `skip_empty_rows`.

If we don't, let's remove it. In which case, the expectation can be come more compact.

Reply:
We don't need it. I just wanted to make sure it wasn't writing an empty line when `vroom_write(data.frame(), file = file, append = TRUE)`

Reply:
I think if `vroom_write(data.frame(), file = file, append = TRUE)` wrote an empty line, that would be a bug and we would want to know about it. So I think the test is better without it (which I see has happened now).

Reply:
Nitpick:

> let's remove it. In which case, the expectation can be come more compact.

There's still this.